graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/AggregationTabularResponseCreator.java

```diff
@@ -75,19 +88,22 @@ private static List<List<Object>> getDatarows(final AggregationRequestSpec searc
                             Collections.nCopies(numGroupings - pivRow.key().size(), "-").stream()  //sometimes pivotRow does not have enough keys - empty value!
                     );
 
-                    final Stream<Object> metrics = searchRequestSpec.metrics().stream()
-                            .map(m -> metricValue(pivRow.values(), m));
+                    final Stream<Object> metrics = series.stream().map(s -> metricValue(s, pivRow.values()));
```

Comment:
We could invoke "pivotRow.values()" only once.

Reply:
done